authors/apps/authentication/backends.py

```diff
@@ -1,13 +1,50 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
-
-"""Configure JWT Here"""
-class JWTAuthentication:
-    def authenticate(self, another):
-        pass
+import jwt
 
+from django.conf import settings
+
+from rest_framework import authentication, exceptions
+
+from .models import User
+
+
+class JWTAuthentication(authentication.BaseAuthentication):
+    """ This validate the token."""
+
+    # initializing the token as an empty string
+    keyword = 'Token'
+
+    def authenticate(self, request):
+        request.user = None
+
+        # returns Authorization header
+        # as a bytestring
+        auth_header = authentication.get_authorization_header(request).split()
+
+        if not auth_header or auth_header[0].lower() != self.keyword.lower():
+            return None
+        if len(auth_header) == 1:
+                raise exceptions.AuthenticationFailed('Invalid token header. No credentials provided.')
+        elif len(auth_header) > 2:
+            raise exceptions.AuthenticationFailed('Invalid token header. Token string should not contain spaces.')
+
+        return self.authenticate_credentials(request, token)
+
+    def authenticate_credentials(self, request, token):
+        """
+        authenticate the given credentials. If authentication is
+        successful, return the user and token. If not, throw an error.
+        """
+        try:
+            payload = jwt.decode(token, settings.SECRET_KEY)
+        except:
```

Comment:
This should be more explicit. What exception are you expecting? This should raise a warning on your `IDE`

Reply:
let me look at it now..